examples/objectives_examples.ipynb

```diff
@@ -0,0 +1,384 @@
+    "\n",
+    "The objective weights are captured in the `Objective` class; for more details, see its [API documentation](https://pyjobshop.org/stable/api/pyjobshop.html#pyjobshop.ProblemData.Objective)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## General calculation of a schedule's objective value"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We now explain how the objective value is calculated in general. To that end, let $w_j$ denote the weight of job $j$ and let $w_{\\text{obj}}$ denote the weight of objective $\\text{obj}$ (see the previous section for an overview of all objectives). We denote the contribution of job $j$ to the objective $\\text{obj}$ in schedule $s$ as $\\text{contribution}(j, \\text{obj}, s)$. The value of the makespan of schedule $s$ is denoted by $\\text{makespan}(s)$. The general objective value for schedule $s$ is calculated in PyJobShop as follows:\n",
+    "\\begin{equation}\n",
+    "\\text{objective}(s) = w_{\\text{makespan}} \\cdot \\text{makespan}(s) + \\sum_{\\text{obj} \\neq \\text{makespan}} w_{\\text{obj}} \\cdot \\sum_{j} w_j \\cdot \\text{contribution}(j, \\text{obj}, s).\n",
+    "\\end{equation}\n",
+    "The following simple example of two jobs, with given weights, release date, and start and end times, shows how the objective components are calculated and combined in the different objective values. In particular, the following table shows the objective component values for the two jobs.\n",
+    "\n",
+    "| **Job** | **Weights** | **Release date** | **Start time** | **End time** | **Due date** | **Tardy? (1=yes, 0=no)** | **Flow time** | **Tardiness** | **Earliness** | **Lateness** |\n",
+    "|:-------:|:-----------:|:----------------:|:--------------:|:------------:|:------------:|:------------------------:|:-------------:|:-------------:|:-------------:|:------------:|\n",
+    "|    1    |      3      |         1        |        2       |       4      |       3      |             1            |       3       |       1       |       0       |       1      |\n",
+    "|    2    |      1      |         0        |        0       |       2      |       4      |             0            |       2       |       0       |       2       |      -2      |\n",
+    "\n",
+    "Based on these objective component values, the following table shows the different objective values for the example.\n",
+    "\n",
+    "| **Objective**            | **Value** |\n",
+    "|--------------------------|:---------:|\n",
+    "| **Makespan**             |     4     |\n",
+    "| **Number of tardy jobs** |     3     |\n",
+    "| **Total flow time**      |     11    |\n",
+    "| **Total tardiness**      |     3     |\n",
+    "| **Total earliness**      |     2     |\n",
+    "| **Maximum tardiness**    |     3     |\n",
+    "| **Maximum lateness**     |     1     |"
```

Comment:
The maximum lateness should be 3. The lateness of job 1 is 1 and the lateness of job 2 is -2. But the maximum lateness $L_\max$ is calculated as $\max_{j \in J} w_j L_j$, where $w_j$ is the job weight, $L_j$ is the lateness of job $j$.

(I haven't explained this in detail anywhere so this is on me)